backend/src/main/java/com/example/dontwastetomuch/game/GameService.java

```diff
@@ -62,27 +80,42 @@ public void addMyGame(MyUser myUser, String gameId){
             myUser.addGameData(gameData);
             myUserRepo.save(myUser);
         } else {
-            throw new IllegalStateException("Game already added to your list");
+            throw new IllegalArgumentException("Game already added");
         }
     }
 
     public void removeMyGame(MyUser myUser, String gameId){
-        GameData data = myUser.getGameData().stream().filter(game -> gameId.equals(game.getGameId())).findAny().orElseThrow();
-        myUser.getGameData().remove(data);
-        myUserRepo.save(myUser);
+        if (myUser.getGameData().stream().anyMatch(game -> gameId.equals(game.getGameId()))){
+            GameData data = myUser.getGameData().stream().filter(game -> gameId.equals(game.getGameId())).findAny().orElseThrow();
+            myUser.getGameData().remove(data);
+            myUserRepo.save(myUser);
+        } else {
+            throw new IllegalArgumentException("Game not found");
+        }
+
+
     }
 
     public List<GameData> getAllMyGames(MyUser myUser) {
-        return myUser.getGameData();
+        if (myUser.getGameData().isEmpty()){
+            throw new NoSuchElementException("Add some games to your list");
+        } else {
+            return myUser.getGameData();
+        }
     }
 
     public void updateGameStats(MyUser user, NewStatsDTO newStatsDTO) {
         GameData gameData = user.getGameData().stream().filter(game -> newStatsDTO.getGameId().equals(game.getGameId())).findAny().orElseThrow();
-        gameData.setPlaytime(gameData.getPlaytime() + newStatsDTO.getAddedPlaytime());
-        gameData.setSpentMoneyGame(gameData.getSpentMoneyGame() + newStatsDTO.getAddedSpentMoneyGame());
-        gameData.setSpentMoneyCoins(gameData.getSpentMoneyCoins() + newStatsDTO.getAddedSpentMoneyCoins());
-        gameData.setSpentMoneyGamePass(gameData.getSpentMoneyGamePass() + newStatsDTO.getAddedSpentMoneyGamePass());
-        myUserRepo.save(user);
+        if (gameData.getGameId().equals(newStatsDTO.getGameId())){
```

Comment:
is the if clause necessary? the checks before (findAny -> orElseThrow) look good. but it wont hurt anyway